src/server-middleware/healthcheck.js

```diff
@@ -1,4 +1,52 @@
-export default function healthcheck(req, res) {
-  res.setHeader('Content-Type', 'text/plain')
-  res.end('OK')
+import osu from 'node-os-utils'
+
+/**
+ * A Healthcheck which passes if the amount of avaliable memory
+ * is greater than our desired minimum avaliable memory percentage.
+ *
+ * Useful in the event of memory leaks, in particular.
+ *
+ * @type {import('@nuxt/types').ServerMiddleware}
+ */
+export default function Healthcheck(_, res) {
+  osu.mem
+    .info()
+    .then((info) => {
+      const avaliableMemory = info.freeMemPercentage
+
+      /**
+       * Server Middleware do not have access to any context or configuration,
+       * so we use a raw environment variable and set a default here, instead
+       * of in `nuxt.config.ts` or `utils/env.ts`.
+       */
+      const desiredAvaliableMemory = Number(
+        process.env.HEALTHCHECK_DESIRED_AVALIABLE_MEMORY_PERCENTAGE ?? '10'
```

Comment:
Can we make the default something that would be sensible for production, rather than the value we want to use for testing? The environment variable should be sufficient for testing with very low values.

Reply:
My intention _was_ to use something sensible for production, I thought 10% would be fine but I'm thinking something closer to 20% will be more reasonable. Do you think even higher, @sarayourfriend?

Reply:
Oh gosh, this is _available_ memory, not utilised memory. My bad! I misread how this worked. 10 or 20, something around there makes perfect sense to me.